src/connectors/star-rating/connectStarRating.js

```diff
@@ -196,8 +196,48 @@ export default function connectStarRating(renderFn, unmountFn) {
         return nextState;
       },
 
+      getWidgetState(fullState, { state }) {
+        const refinedStar = this._getRefinedStar(state);
+        if (
+          refinedStar === undefined ||
+          (fullState &&
+            fullState.starRating &&
+            fullState.starRating[attributeName] === refinedStar)
+        )
+          return fullState;
```

Comment:
Could we wrap this condition with brackets?

Reply:
Yes we can!